src/contributing/adding_new_operation_how_to.md

```diff
@@ -0,0 +1,108 @@
+# Adding operations to Parsec
+
+This mini how-to will cover how to add operations to the backend and frontend of Parsec. The steps
+for adding operations to a backend will differ between providers, this guide will cover Mbed Crypto.
+The steps for adding operations to the frontend should be the same for all backend providers.
+
+## Backend (Mbed Crypto)
+
+### psa-crypto-sys
+
+Locate the parts of the PSA API in `mbedtls` that will need to be used during the operation you are
+adding. This includes all constants, datatypes, macros and functions. Note: `mbedtls` does not yet
+fully conform to the PSA API 1.0 specification, so there may be inconsistencies between the PSA
+documentation and the macros and functions that `mbedtls` exposes.
+
+Starting in `psa-crypto-sys`, constants are added to `constants.rs`. Inline functions and macros
+require a shim. Add the inline function and macro definitions into `shim.h`, `shim.c` and
+`shim_methods.rs`. When adding entries to `shim_methods.rs`, take note of where you place `unsafe`.
+If the PSA API specification states that under any circumstance, an unspecified result can be
+returned, then mark the entire `fn` definition as `unsafe`. Otherwise, if the return values are
+always specified, wrap the call to the shim method as `unsafe`. Regular functions are added to `pub
+use psa_crypto_binding::{ … }`.
+
+### psa-crypto
+
+Define a Rust native interface for the operation you are adding. The interface will work between the
+Rust native datatypes given to it and the C bindings in `psa-crypto-sys`. It is important to ensure
+that the Rust interface handles all possible situations (e.g. closing a key-handle on an error) to
+ensure it is a safe Rust native interface to the operation.
+
+At this point, you will now have a safe Rust interface for the PSA operation you added.
+
+## Frontend (Parsec)
+
+### parsec-book
+
+Create the specification page for the Parsec Book. Include operation details such as parameters,
```

Comment:
I know this is annoying but could you add links of the different pages you are referring to 😬 (or examples of those)? So that it is easier for readers to just click on the page instead of searching themselves.

Reply:
Yes no problem! This is still in draft mode on purpose!